apps/dhparam.c

```diff
@@ -219,16 +232,28 @@ int dhparam_main(int argc, char **argv)
         } else
 # endif
         {
-            dh = DH_new();
             BIO_printf(bio_err,
                        "Generating DH parameters, %d bit long safe prime, generator %d\n",
                        num, g);
             BIO_printf(bio_err, "This is going to take a long time\n");
+            if (!EVP_PKEY_paramgen_init(ctx)) {
+                BN_GENCB_free(cb);
+                ERR_print_errors(bio_err);
+                goto end;
+            }
+            if (!EVP_PKEY_paramgen(ctx, &pkey)) {
+                BN_GENCB_free(cb);
+                ERR_print_errors(bio_err);
+                goto end;
+            }
+            /*
```

Comment:
Do we want to keep the old code in comment? I do not think so.

Reply:
No, old code left for reference that I missed.